  0%|          | 0/100000 [00:00<?, ?it/s]/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/autograd/graph.py:769: UserWarning: Error detected in LinearBackward0. Traceback of forward call that caused the error:
  File "/Users/hamidosooli/PycharmProjects/MARL/COMA/training.py", line 99, in <module>
    targets, critic_value, advantage = alg.target_lambda_return()
  File "/Users/hamidosooli/PycharmProjects/MARL/COMA/alg.py", line 112, in target_lambda_return
    critic_action_value = self.critic_net(critic_inp)
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/hamidosooli/PycharmProjects/MARL/COMA/network.py", line 17, in forward
    q = self.layer3(x)
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1553, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1562, in _call_impl
    return forward_call(*args, **kwargs)
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/nn/modules/linear.py", line 117, in forward
    return F.linear(input, self.weight, self.bias)
 (Triggered internally at /Users/runner/work/pytorch/pytorch/pytorch/torch/csrc/autograd/python_anomaly_mode.cpp:116.)
  return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
  0%|          | 0/100000 [00:01<?, ?it/s]
Traceback (most recent call last):
  File "/Users/hamidosooli/PycharmProjects/MARL/COMA/training.py", line 124, in <module>
    actors_grad_norm = alg.optimize_actors()
  File "/Users/hamidosooli/PycharmProjects/MARL/COMA/alg.py", line 69, in optimize_actors
    self.actors_loss.backward()
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/_tensor.py", line 521, in backward
    torch.autograd.backward(
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/autograd/__init__.py", line 289, in backward
    _engine_run_backward(
  File "/Users/hamidosooli/PycharmProjects/pythonProject1/venv/lib/python3.9/site-packages/torch/autograd/graph.py", line 769, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [MPSFloatType [125, 128]] is at version 2; expected version 1 instead. Hint: the backtrace further above shows the operation that failed to compute its gradient. The variable in question was changed in there or anywhere later. Good luck!
